### PR TITLE
APPS/cmp: improve diagnostics for presence of TLS options

### PR DESCRIPTION
**Pull request by DDvO**
- [x] documentation is added or updated

### CODE REVIEW

**Comment by DDvO** [comment]
How about backporting this to 3.0?
This is (just) about user guidance.

**Comment by t8m** [comment]
> How about backporting this to 3.0?
> This is (just) about user guidance.

This is borderline. In my personal opinion this should not go into 3.0.

**Comment by DDvO** [comment]
> This is borderline. In my personal opinion this should not go into 3.0.

Fine by me.
Anyway, since you apparently already had a look, can you approve this for master?

**Review by t8m — APPROVED** [review]
OK for master.

File: apps/cmp.c [hunk 1]
```diff
@@ -1312,13 +1315,18 @@ static SSL_CTX *setup_ssl_ctx(OSSL_CMP_CTX *ctx, const char *host,
...
         EVP_PKEY_free(pkey); /* we do not need the handle any more */
+    } else {
+        CMP_warn("-tls_used given without -tls_key; cannot authenticate to the TLS server");
     }
-    if (opt_tls_trusted != NULL) {
-        /* enable and parameterize server hostname/IP address check */
+    if (trust_store != NULL) {
+        /*
+         * Enable and parameterize server hostname/IP address check.
+         * If we did this before checking own TLS cert
```
Comment (paulidale) [inline apps/cmp.c:1324]: _checking **our** own TLS cery_

**Review by paulidale — APPROVED** [review]
One minor wording suggestion.

File: apps/cmp.c (hunk 1 above)
Comment (DDvO) [reply apps/cmp.c]: Pushed fixup for this.

**Comment by t8m** [comment]
Still OK, no need to restart the waiting period.

**Comment by DDvO** [comment]
Merged - thanks @t8m and @paulidale

**Comment by t8m** [comment]
It is unclear to me, where this got pushed. It seems there was some mistake done when merging this? @DDvO  Reopening.

**Comment by DDvO** [comment]
Like with #17358, rebased now on `upstream/master` and force-pushed.
Looks good now.

**Comment by DDvO** [comment]
Now it's really on the openssl master. Sorry for the confusion.
I had mixed it up with the master of my fork.

**Pull request closed without merging**